eno-core/src/main/java/fr/insee/eno/core/processing/out/steps/lunatic/LunaticAddControlMandatory.java

```diff
@@ -0,0 +1,108 @@
+     */
+    public LunaticAddControlMandatory(EnoParameters.Language language) {
+        switch (language) {
+            case FR, EN -> this.language = language;
+            case null -> this.language = EnoParameters.Language.FR; // null => default
+            default -> this.language = EnoParameters.Language.EN; // non-translated cases => english
+        }
+    }
+
+    /** Generates "mandatory" controls for components that have the "mandatory" attribute set to true. */
+    @Override
+    public void apply(Questionnaire lunaticQuestionnaire) {
+        addMandatoryControls(lunaticQuestionnaire.getComponents());
+        lunaticQuestionnaire.getComponents().stream().filter(Loop.class::isInstance)
+                .map(Loop.class::cast)
+                .map(Loop::getComponents)
+                .forEach(this::addMandatoryControls);
+    }
+
+    private void addMandatoryControls(List<ComponentType> lunaticComponents) {
+        lunaticComponents.stream()
+                .filter(ComponentSimpleResponseType.class::isInstance)
+                // use the "simple response" interface to get the mandatory prop
+                // TODO: having an interface for "mandatory" in Lunatic-Model would be better
+                .map(ComponentSimpleResponseType.class::cast)
+                .filter(ComponentSimpleResponseType::getMandatory)
+                // re-cast as a Lunatic component object
+                .map(ComponentType.class::cast)
+                .forEach(this::addMandatoryControl);
+    }
+
+    private void addMandatoryControl(ComponentType lunaticComponent) {
+        Optional<String> expression = generateMandatoryControlExpression(lunaticComponent);
+        if (expression.isEmpty())
+            return;
+        lunaticComponent.getControls().addFirst(
+                createMandatoryControl(lunaticComponent.getId() + "-mandatory-check", expression.get()));
+    }
+
+    /** Generates a "mandatory" control expression in function of the component type.
+     * If the component is not concerned by the "mandatory" property, the returned value is empty. */
+    private Optional<String> generateMandatoryControlExpression(ComponentType lunaticComponent) {
+        if (lunaticComponent.getComponentType() == null)
+            throw new IllegalArgumentException("Component " + lunaticComponent.getId() + " has no type defined.");
+        String responseName = ((ComponentSimpleResponseType) lunaticComponent).getResponse().getName();
+        return switch (lunaticComponent.getComponentType()) {
+            case INPUT, TEXTAREA ->
+                    Optional.of(String.format("not(nvl(%s, \"\") = \"\")", responseName));
+            case INPUT_NUMBER,
+                    DATEPICKER,
+                    DURATION,
+                    RADIO, DROPDOWN, CHECKBOX_ONE,
+                    CHECKBOX_BOOLEAN ->
+                    Optional.of(String.format("not(isnull(%s))", responseName));
+            default -> Optional.empty();
```

Comment:
pour les switchs pour un enum, je trouve préférable d'éviter le cas par défaut : je trouve préférable de traiter explicitement toutes les valeurs pour éviter les oublis is l'enum bouge